### PR TITLE
Add method to client to check login is established

### PR DESCRIPTION
**Pull request by sandrovicente**
Method Client.login() will throw and LoginError exception if Exchange doesn't accept it.
Alternatively, client postpone 'read()' upon login using 'login(False)'. In this case, a call 'client.check_login()' should follow a call to 'client.read()'

### CODE REVIEW

File: smarkets/streaming_api/client.py [hunk 1]
```diff
@@ -48,19 +48,43 @@ def __init__(self, session):
...
+            self.check_login()
             self.flush()
 
+    def check_login(self):
+        """
+        Check if login was successful.
+        Must be executed after a 'read()' and throws an exception if not successful
+        """
+        if not self.last_login:
+            raise LoginError(eto.LOGOUT_NONE)
```
Comment (jcpetruzza) [inline smarkets/streaming_api/client.py]: @sandrovicente I'm wondering if we should raise a `LoginTimeOut` or something here instead. The rationale is that users catching this exception may be confused thinking that a `logout` message was received when the actual issue would be that no such message was received.

File: smarkets/streaming_api/session.py [hunk 2]
```diff
@@ -132,6 +134,8 @@ def logout(self):
     def disconnect(self):
         "Disconnects from the API"
         self.socket.disconnect()
+        self.inseq = self.init_inseq
+        self.outseq = self.init_outseq
```
Comment (jcpetruzza) [inline smarkets/streaming_api/session.py]: @sandrovicente Why is this required?

File: smarkets/streaming_api/client.py (hunk 1 above)
Comment (sandrovicente) [reply smarkets/streaming_api/client.py]: Ok. Changed to throw 'LoginTimeout'

File: smarkets/streaming_api/session.py (hunk 2 above)
Comment (sandrovicente) [reply smarkets/streaming_api/session.py]: inseq and outseq may be given as init parameters. I save the ones received to 'init_XXX' so in case of reconnection I keep using them. Not really sure of a use case for this though, only to keep consistent

**Comment by jcpetruzza** [comment]
LGTM. Waiting for CI issues to be fixed
